src/app/ir/conv.ts

```diff
@@ -0,0 +1,73 @@
+import { InOutNode, ModelNode } from '@/app/ir/mainNodes';
+import { Option, UUID } from '@/app/util';
+import {
+  ActivationF, Initializer, Padding, Regularizer,
+} from '@/app/ir/irCommon';
+
+abstract class Conv implements ModelNode, InOutNode {
+  abstract readonly outputs: Set<UUID>;
+
+  abstract readonly input: UUID;
+
+  abstract readonly filters: bigint;
+
+  abstract readonly padding: Padding;
+
+  abstract readonly weights: [Initializer, Regularizer]
```

Comment:
I think we should be more explicit here with `weightsInitializer` and `weightsRegularizer`
It can be misleading and we want to be futureproof, maybe we might want to load in the actual weights some day, same for biases

Reply:
but we are doing interop with ML frameworks right, so the initialiser is a function that provides weights. We will provide it at some point

I don't see your point